### PR TITLE
Add store icon

### LINKED ISSUE #23835: Add a store icon
A store icon would be handy for a project we're working on at WooCommerce, and in other applications I am sure.

Before opening a PR I wanted to kick off discussion around some initial design concepts.

We already have the [Home](https://github.com/WordPress/gutenberg/blob/master/packages/icons/src/library/home.js) icon in the set, so it makes sense to build upon that and ensure any new store icon feels like it lives in the same city as the home icon (as my pal @jasmussen so eloquently put it :D)

I borrowed concepts from the Noun Project to put these options together:

store

Option 1 is the simplest and most widely-used interpretation I think - a building with an awning and a door. It's also the most minimal execution I could come up with, which aligns with a core principle of this icon set. Options 2 and 3 are more expressive, seeking to solidify the metaphor in case option 1 doesn't clearly communicate it. Personally I think option 1 does the job, but would be keen to hear thoughts, or see alternative ideas before getting in to the code and polishing the designs.

Edit: Figma link [here](https://www.figma.com/file/fnyj380i05vGzuuH60frLQ/G2-Design?node-id=3234%3A24162) in case anyone would like to riff on these sketches.

### PR DESCRIPTION
**Pull request by jameskoster**
Closes #23835

Adds a new "Store" icon to the Icons package, based on the existing icon for "Home":

Figma [here](https://www.figma.com/file/fnyj380i05vGzuuH60frLQ/G2-Design?node-id=3234%3A24162).

### CODE REVIEW

File: packages/icons/src/library/store.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/**
+ * WordPress dependencies
+ */
+import { Path, SVG } from '@wordpress/primitives';
+
+const store = (
+	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+		<Path
+			fillRule="evenodd"
```
Comment (jasmussen) [inline packages/icons/src/library/store.js:9]: I'm assuming both this and the cliprule are needed for this to work? I only see a single Path so it must be a compound shape, I just always get confused when I see these exotic properties.

Comment (jameskoster) [reply packages/icons/src/library/store.js:9]: To be honest, so do I. I always remove them and test the output to be sure they are required. In this case, sections of the "awning" in the icon become filled unless the property is included.

**Review by jasmussen — APPROVED** [review]
This is a good design, and the store looks like it exists in the same city as the house. 👍 👍

As we add icons to the component, it seems to make sense to lay down some ground rules for what goes into the compent and what doesn't. Notably this PR simply adds the icon to the component, it doesn't also use it. So it begs the question what goes in and what doesn't.

One benefit of an icon component like this is it enables a consistency across projects using these, instead of having to do a mix and match of various things. So it feels like it would be okay to add some icons, even if they aren't also immediately used by the block editor. So let's lay down some ground rules, and we can revisit, refine, and iterate these as we go. Suggestion:

- If the icon is needed in the project, it should go in as fast as can be reviewed.
- If the icon is not used in the project, the icon should be generic in subject and in the interest of (and free for) 3rd parties to use. In this case, the PR should gather some design feedback (via the label) and then be added if seen as useful.

In that vein, I can see a ton of folks wanting a store icon. Let's get this in.

**Comment by jasmussen** [comment]
Thank you Jay.

**Pull request merged**